src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -21,25 +19,30 @@ public class IPAddress
 
         internal const long LoopbackMask = 0x00000000000000FF;
 
-        // IPv6 Changes: make this internal so other types that understand about
-        //               IPv4 and IPv4 can still access it rather than the obsolete property.
-        internal long Address;
-
-        internal string StringRepresentation;
-
         public static readonly IPAddress IPv6Any = new IPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, 0);
         public static readonly IPAddress IPv6Loopback = new IPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 }, 0);
         public static readonly IPAddress IPv6None = new IPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, 0);
 
-        /// <devdoc>
-        ///   <para>
-        ///     Default to IPv4 address
-        ///   </para>
-        /// </devdoc>
-        private readonly AddressFamily _family = AddressFamily.InterNetwork;
-        private readonly ushort[] _numbers = new ushort[NumberOfLabels];
-        private long _scopeId = 0;
-        private int _hashCode = 0;
+        /// <summary>
+        /// For IPv4 addresses, this field stores the Address.
+        /// For IPv6 addresses, this field stores the ScopeId.
```

Comment:
This is a definite no-go. Code clarity and maintainability is more important than the few bytes that you've saved.

There are a lot of other options if you wish to save the bytes:
- create different subclasses for ipv4/v6 (preferred for code clarity)
- use an _union_ style structure (see [FieldOffsetAttribute](https://msdn.microsoft.com/en-us/library/system.runtime.interopservices.fieldoffsetattribute(v=vs.110).aspx))